File: haproxy/src/main/java/org/kevoree/library/java/haproxy/Reader.java
package org.kevoree.library.java.haproxy;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 04/12/2013
 * Time: 19:11
 */
public class Reader implements Runnable {

    private BufferedReader br;

    public Reader(InputStream i) {
        this.br = new BufferedReader(new InputStreamReader(i));
    }

    @Override
    public void run() {
        try {
            String value;
            while ((value = br.readLine()) != null) {
                System.out.println(value);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: cloud/org.kevoree.library.cloud.lightlxc/src/main/java/org/kevoree/library/cloud/lightlxc/wrapper/Reader.kt
package org.kevoree.library.cloud.lightlxc.wrapper

import java.io.InputStream
import java.io.BufferedReader
import java.io.InputStreamReader
import java.io.IOException

/**
 * Created by root on 25/02/14.
 */
class Reader(inputStream: InputStream, val nodeName: String, val error: Boolean) : Runnable{
    val br: BufferedReader
    {
        br = BufferedReader(InputStreamReader(inputStream));
    }
   override fun run() {
        var line: String?;
        try {
            line = br.readLine()
            while (line != null) {
                line = nodeName + "/" + line
                if (error) {
                    System.err.println(line);
                } else {
                    System.out.println(line);
                }
                line = br.readLine()
            }
        } catch (e: IOException) {
            e.printStackTrace();
        }
    }
}